### PR TITLE
Implement basic offence information endpoint

### PR DESCRIPTION
**Pull request by AprilMD**
Added an initial offences endpoint. Notably:
- Offence histories only currently come from Prison API.
- We've exposed 3 fields only: date, offence code, and description. These will be expanded in line with requirements.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/models/Offence.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package uk.gov.justice.digital.hmpps.hmppsintegrationapi.models
+
+import java.time.LocalDate
+
+data class Offence(
+  val date: LocalDate?,
+  val code: String?,
+  val description: String?,
+)
```
Comment (mcoffey-mt) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/models/Offence.kt:9]: I think these can be non-nullable based on the fact that we're only pulling from NOMIS right now, and these fields in NOMIS are not nullable

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/controllers/v1/person/OffencesController.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.Offence
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.UpstreamApi
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.UpstreamApiError
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.services.GetOffencesForPersonService
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.util.PaginatedResponse
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.util.paginateWith
+
+@RestController
+@RequestMapping("/v1/persons")
+class OffencesController(
```
Comment (chubberlisk) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/controllers/v1/person/OffencesController.kt:20]: Really like this approach of putting this under a `controllers/v1/person` package, makes a controller smol!

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/gateways/nomis/GetOffencesForPersonTest.kt [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+        Mockito.reset(hmppsAuthGateway)
+        whenever(hmppsAuthGateway.getClientToken("NOMIS")).thenReturn(HmppsAuthMockServer.TOKEN)
+      }
+
+      afterTest {
+        nomisApiMockServer.stop()
+      }
+
+      describe("#getOffences") {
```
Comment (chubberlisk) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/gateways/nomis/GetOffencesForPersonTest.kt:74]: This `describe` is redundant now that each file only tests one gateway method.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/models/Offence.kt (hunk 1 above)
Comment (emileswarts) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/models/Offence.kt:9]: Agreed

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/models/nomis/OffenceTest.kt [hunk 4]
```diff
@@ -0,0 +1,25 @@
+package uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.nomis
+
+import io.kotest.core.spec.style.DescribeSpec
+import io.kotest.matchers.shouldBe
+import java.time.LocalDate
+
+class OffenceTest : DescribeSpec(
```
Comment (chubberlisk) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/models/nomis/OffenceTest.kt:7]: This should be `OffenceHistoryDetailTest` and the file name should be the same.

**Review by emileswarts — APPROVED** [review]

**Review by chubberlisk — APPROVED** [review]
Very noice work - like the setup with the controller and well done for remembering to update the OpenAPI spec too!

**Pull request merged**